src/pages/utils/data.ts

```diff
@@ -91,16 +95,33 @@ export const mergeFileUploadErrors = (
     return errors;
   }, []);
 
-export const mergeAttributesWithFileUploadResult = (
+export const getAttributesFromFileUploadResult = (
   attributesWithNewFileValue: FormsetData<null, File>,
   uploadFilesResult: Array<MutationFetchResult<FileUpload>>
-): AttributeValueInput[] =>
-  uploadFilesResult.map((uploadFileResult, index) => {
-    const attribute = attributesWithNewFileValue[index];
-
-    return {
-      file: uploadFileResult.data.fileUpload.uploadedFile.url,
+): AttributeValueInput[] => {
+  const removedFileValues = attributesWithNewFileValue
+    .filter(attribute => !attribute.value)
+    .map(attribute => ({
+      file: undefined,
       id: attribute.id,
       values: []
-    };
-  }, []);
+    }));
+
+  const fileAttributesToUpload = attributesWithNewFileValue.filter(
```

Comment:
Isn't this part, as well as `uploadedFileValues` used also in some other places? I think it could be reused, instead of rewritten